src/svelte/Breadcrumbs.svelte

```diff
@@ -1,21 +1,99 @@
 <script lang="ts">
-  interface Breadcrumbs {
-    items: Array<{url: string; title: string}>;
-    ariaLabel?: string;
-  }
+  import {onMount} from "svelte";
+  import type {Breadcrumbs} from "../ts/types";
+  import {useMachine} from '@xstate/svelte';
+  import {createMachine} from 'xstate';
 
+  const BUTTON_ELLIPSIS = Symbol()
   export let breadcrumbs: Breadcrumbs = {items: []};
-  let ariaLabel = breadcrumbs.ariaLabel ?? 'breadcrumbs';
+  export let loadJs = false
+
+  interface BreadcrumbsContext {
+    breadcrumbsItems: Array<{url: string, title: string} | symbol>
+  }
+
+  type BreadcrumbsEvent = {type: 'MOUNTED'} | {type: 'TOGGLE'};
+
+  type BreadcrumbsState =
+    | {value: 'serverRendered'; context: BreadcrumbsContext}
+    | {value: 'closed'; context: BreadcrumbsContext}
+    | {value: 'open'; context: BreadcrumbsContext};
+
+  const breadcrumbsMachine = createMachine<BreadcrumbsContext, BreadcrumbsEvent, BreadcrumbsState>({
+    id: 'breadcrumbs',
+    initial: 'serverRendered',
+    context: {
+      breadcrumbsItems: []
+    },
+    states: {
+      serverRendered: {
+        on: {
+          MOUNTED: [
+            {
+              target: 'closed',
+              cond: (() => breadcrumbs.items.length > 3)
+            },
+            {
+              target: 'open'
+            }
+          ]}
+      },
+      closed: {
+        entry: 'prepareCollapsedBreadcrumbs',
+        on: { TOGGLE: 'open' }
+      },
+      open: {
+        entry: 'expandBreadcrumbs',
+        on: { TOGGLE: 'closed' }
+      }
+    }
+  }, {
+    actions: {
+      prepareCollapsedBreadcrumbs: (context) => {
+        context.breadcrumbsItems = [
+          breadcrumbs.items[0],
+          breadcrumbs.items[1],
+          BUTTON_ELLIPSIS,
+          breadcrumbs.items[breadcrumbs.items.length -1]]
+      },
+      expandBreadcrumbs: (context) => {
+        context.breadcrumbsItems = breadcrumbs.items
+      }
+    }
+  })
+
+  const {state, send} = useMachine(breadcrumbsMachine)
+
+  let ariaLabel = breadcrumbs.ariaLabel ?? 'brødsmulesti';
+  let showAllBreadCrumbs = breadcrumbs.showAllAriaLabel ?? 'vis mer'
```

Comment:
Maybe use a name that clarifies that this is a string meant as a label?

```suggestion
  let showAllBreadCrumbsLabel = breadcrumbs.showAllAriaLabel ?? 'vis mer'
```